src/lualdap.c

```diff
@@ -822,9 +792,22 @@ static int get_attrs_param (lua_State *L, int idx, char *attrs[]) {
 		attrs[1] = NULL;
 	} else if (type != LUA_TTABLE)
 		attrs[0] = NULL;
-	else
-		if (table2strarray (L, lua_gettop (L), attrs, LUALDAP_MAX_ATTRS))
-			return 0;
+	else {
+		int i;
+		int n = lua_rawlen(L, -1);
+		if (LUALDAP_MAX_ATTRS < (n+1))
```

Comment:
Since you are touching this already: `n >= LUALDAP_MAX_ATTRS` appears to be clearer.